### ISSUE TITLE
[AIR] Don't unsqueeze feature and label tensors in `Dataset.to_torch`  by default 

### ISSUE THREAD

**Issue by bveeramani** [ray-project/ray#25146]
### Description

Change 

```
def to_torch(
        ...
        unsqueeze_label_tensor: bool = True,
        unsqueeze_feature_tensors: bool = True,
    ) 
```

to

```
def to_torch(
        ...
        unsqueeze_label_tensor: bool = False,
        unsqueeze_feature_tensors: bool = False,
    ) 
```

or remove the two parameters altogether.

Also see #24973 and #25145 

### Use case

Ideally, default values should work for common use cases,  but you can't train with image data without setting the `unsqueeze_feature_tensors` and `unsqueeze_label` parameters.

```
train_dataset_shard: torch.utils.data.Dataset = train.get_dataset_shard("train").to_torch(
        feature_columns=["image"],
        label_column="label",
        batch_size=config["batch_size"],
        unsqueeze_feature_tensors=False,
        unsqueeze_label_tensor=False
    )
```

**Comment by ericl** [comment]
Downgrading priority since to_torch/to_tf will be experimental in 2.0 (we should make sure to mark them as so).
